File: leetcode/2125. Number of Laser Beams in a Bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        if (bank.size() < 2)
            return 0;

        int ans = 0, lastRowOnes = 0;
        for (string row : bank) {
            int countOnes = 0;
            for (char c : row)
                countOnes += c - 48;

            ans += lastRowOnes * countOnes;
            lastRowOnes = (countOnes == 0) ? lastRowOnes : countOnes;
        }
        return ans;
    }
};

File: leetcode/2125. Number of Laser Beams in a Bank.py
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        floor_count = [
            floor.count("1") for floor in bank if floor.count("1") > 0
        ]
        return sum(
            [
                floor_count[i] * floor_count[i + 1]
                for i in range(len(floor_count) - 1)
            ]
        )
